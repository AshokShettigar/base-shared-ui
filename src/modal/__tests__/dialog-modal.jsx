import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BUTTON_VARIATION } from 'platform-common-ui';
import DialogModal, { Dialog } from 'platform-common-ui/modal/dialog-modal';

describe('modal/dialog-modal', function() {
    const mockStore = configureStore();

    it('should render the component', function() {
        const handler = () => true;
        const controls = [{
            label: 'OK',
            onClick: handler,
            variation: BUTTON_VARIATION.PRIMARY
        }];

        const wrapper = shallow(
            <Dialog
                formControls={controls}
                id="alert1"
            >
                Alert description
            </Dialog>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the component with title', function() {
        const handler = () => true;
        const controls = [{
            label: 'OK',
            onClick: handler,
            variation: BUTTON_VARIATION.PRIMARY
        }];

        const wrapper = shallow(
            <Dialog
                formControls={controls}
                id="alert1"
                title="Alert title"
            >
                Alert description
            </Dialog>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the component with height and width', function() {
        const handler = () => true;
        const controls = [{
            label: 'OK',
            onClick: handler,
            variation: BUTTON_VARIATION.PRIMARY
        }];

        const wrapper = shallow(
            <Dialog
                formControls={controls}
                height="100px"
                id="alert1"
                width="100px"
            >
                Alert description
            </Dialog>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the component', function() {
        const handler = () => true;
        const controls = [{
            label: 'OK',
            onClick: handler,
            variation: BUTTON_VARIATION.PRIMARY
        }];

        const wrapper = mount(
            <Provider store={mockStore()}>
                <DialogModal
                    formControls={controls}
                    id="alert1"
                    title="Alert title"
                >
                    Alert description
                </DialogModal>
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
