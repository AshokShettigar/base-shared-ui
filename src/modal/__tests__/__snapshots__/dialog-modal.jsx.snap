// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal/dialog-modal should render the component 1`] = `
<BaseModal
  height="150px"
  id="alert1"
  title="Alert"
  width="450px"
>
  <div
    className="cn-modal__content"
    data-component="Dialog"
  >
    Alert description
  </div>
  <ModalControls
    controls={
      Array [
        Object {
          "label": "OK",
          "onClick": [Function],
          "variation": 1,
        },
      ]
    }
  />
</BaseModal>
`;

exports[`modal/dialog-modal should render the component 2`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(HigherOrderPopupComponent)
    formControls={
      Array [
        Object {
          "label": "OK",
          "onClick": [Function],
          "variation": 1,
        },
      ]
    }
    id="alert1"
    layoverClassName="cn-modal-layover"
    popupClassName="cn-modal cn-alert-modal"
    title="Alert title"
  >
    <HigherOrderPopupComponent
      formControls={
        Array [
          Object {
            "label": "OK",
            "onClick": [Function],
            "variation": 1,
          },
        ]
      }
      id="alert1"
      layoverClassName="cn-modal-layover"
      popupClassName="cn-modal cn-alert-modal"
      title="Alert title"
      updatePopupProps={[Function]}
    />
  </Connect(HigherOrderPopupComponent)>
</Provider>
`;

exports[`modal/dialog-modal should render the component with height and width 1`] = `
<BaseModal
  height="100px"
  id="alert1"
  title="Alert"
  width="100px"
>
  <div
    className="cn-modal__content"
    data-component="Dialog"
  >
    Alert description
  </div>
  <ModalControls
    controls={
      Array [
        Object {
          "label": "OK",
          "onClick": [Function],
          "variation": 1,
        },
      ]
    }
  />
</BaseModal>
`;

exports[`modal/dialog-modal should render the component with title 1`] = `
<BaseModal
  height="150px"
  id="alert1"
  title="Alert title"
  width="450px"
>
  <div
    className="cn-modal__content"
    data-component="Dialog"
  >
    Alert description
  </div>
  <ModalControls
    controls={
      Array [
        Object {
          "label": "OK",
          "onClick": [Function],
          "variation": 1,
        },
      ]
    }
  />
</BaseModal>
`;
